src/JsonLd/ContextBuilder.php

```diff
@@ -97,6 +97,7 @@ public function getResourceContext(string $resourceClass, int $referenceType = U
             }
 
             $convertedName = $this->nameConverter ? $this->nameConverter->normalize($propertyName) : $propertyName;
+            $jsonldContext = $propertyMetadata->getAttributes()['jsonld_context'] ?? [];
```

Comment:
What's set in the `jsonld_context` attribute should be used as the base. We don't want to limit it to just `@type`.